lib/adagrams.rb

```diff
@@ -0,0 +1,98 @@
+require "pry"
+
+#Wave 1
+#method that represent the hand of letters that the player has drawn
+def draw_letters
+  letters = %W[A A A A A A A A B B C C D D D D E E E E E E E E E E E E F F G G G H H I I I I I I I I I J K L L L L M M N N N N N N O O O O O O O O P P Q R R R R R R S S S S T T T T T T U U U U V V W W X Y Y Z]
+  hand = []
+  10.times do
+    #drawing 10 letters at random for user hand
+    index = rand(letters.length)
+    #delete letters that are used in hand to avoid repeats
+    hand << letters.delete_at(index)
+  end
+    #output user hand
+  return hand
+end
+
+#Wave 2
+#method to check if user input only includes letters from the hand
+def uses_available_letters?(input,letters_in_hand)
+  #splitting the letters from user input ["ADA"] = ["A", "D", "A"]
+  #confirming that letters in hand are matching the letters drawn by user
+  input.split("").each do |letter|
+    if letters_in_hand.include?(letter)
+      letters_in_hand.each_with_index do |hand_letter,index|
+        if hand_letter == letter
```

Comment:
This works, it's a little awkward.  You can use a method to get the 1st index of a given argument instead of using the loop.